src/xwayland/xserver.rs

```diff
@@ -80,37 +76,46 @@ pub struct XWayland<Data> {
 /// disconnecting several time in a row, but only a single connection will
 /// be active at any given time.
 #[derive(Debug)]
+#[must_use = "Connection events must be handled to prevent fd leaking"]
```

Comment:
Huh? "Leak" sounds like calling `std::mem::forget` on e.g. a `UnixStream`: A file descriptor ends up being unowned.

Is this really that kind of leak? Shouldn't `UnixStream`'s `Drop` impl clean that up?

Put differently: I (apparently) do not understand the reason text.

Reply:
Didn't know that UnixStream had a Drop implementation (even if you need to go like 8 structs deep in std)

Reply:
Well yeah, it's more like "this event needs to be handled for XWayland to work at all" tbh